main.py

```diff
@@ -0,0 +1,53 @@
+from copy import deepcopy
+
+def primary_forbidden_sequences(list_of_all_forbidden_sequences,P):
```

Comment:
NIT:
`list_of_all_forbidden_sequences` -> `forbidden_sequences` (naming it as a plural is often clear enough)?
Don't forget the space after the coma :p